src/cryptonote_basic/account.cpp

```diff
@@ -67,13 +67,13 @@ DISABLE_VS_WARNINGS(4244 4345)
     m_keys.m_multisig_keys.clear();
   }
   //-----------------------------------------------------------------
-  crypto::secret_key account_base::generate(const crypto::secret_key& recovery_key, bool recover, bool two_random)
+  crypto::secret_key account_base::generate(const crypto::secret_key& recovery_key, bool recover, bool two_random, bool from_legacy16B_lw_seed)
   {
     crypto::secret_key first = generate_keys(m_keys.m_account_address.m_spend_public_key, m_keys.m_spend_secret_key, recovery_key, recover);
 
     // rng for generating second set of keys is hash of first rng.  means only one set of electrum-style words needed for recovery
     crypto::secret_key second;
-    keccak((uint8_t *)&m_keys.m_spend_secret_key, sizeof(crypto::secret_key), (uint8_t *)&second, sizeof(crypto::secret_key));
+    keccak((uint8_t *)&(from_legacy16B_lw_seed == false ? m_keys.m_spend_secret_key : first), sizeof(crypto::secret_key), (uint8_t *)&second, sizeof(crypto::secret_key));
```

Comment:
two reasons. I want the preferred supported primary path to be first (emphasize other cases are secondary) and calling out == false is just one of the things I have taken to doing for explicitness (i dont do it with true) and because ! is easy to miss

Reply:
I don't think this will actually help improve the readability. The order of the conditional doesn't represent things being primary or secondary.  Just doing

```c++
&(from_legacy16B_lw_seed ? first : m_keys.m_spend_secret_key)
```

is shorter and simpler.

Reply:
I see your point. Will do